### ISSUE TITLE
Coverage is not recorder for some projects

### ISSUE THREAD

**Issue by slapikv** [OpenCover/opencover#1008]
Please provide the following information when submitting an issue. 
> Where appropriate replace the `[]` with a `[X]`

**NOTE** Support requests should initially be raised on [GitHub discussions](https://github.com/OpenCover/opencover/discussions) 

### My Framework

- [ ] .NET 2
- [ ] .NET 3.5
- [ ] .NET 4
- [ ] .NET 4.5
- [ ] .NET 4.6
- [ ] .NET 4.6.1
- [ ] .NET 4.6.2
- [ ] .NET 4.7
- [x] .NET 4.7.1
- [ ] .NET 4.7.2
- [ ] .NET 4.8
- [ ] .NET Core 2.2.0
- [ ] .NET Core 3.1.0
- [ ] .NET 5
- only release and LTS versions of .NET runtimes and SDKs will be supported 

### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL)
- [ ] Windows 8
- [ ] Windows 8.1
- [x] Windows 10
- [ ] Windows 10 IoT Core
- [ ] Windows Server 2012
- [ ] Windows Server 2012 R2
- [ ] Windows Server 2016
- [ ] Windows Server 2019

### I have already...

- [x] repeated the problem using the latest stable release of OpenCover.
- [x] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf).
- [x] have looked at the opencover output xml file in an attempt to resolve the issue.
- [x] reviewed the [current issues](https://github.com/opencover/opencover/issues) to check that the issue isn't already known.

### My issue is related to (check only those which apply):

- [x] no coverage being recorded
- [ ] 32 or 64 bit support

### Describe the bug
I am using Cake for Opencover execution, the test framework is xUnit. The solution has more than 10 projects it is [Sitecore Heilx](https://helix.sitecore.com/principles/visual-studio/solutions.html) based. The issue is that some of the projects are covered correctly, some of them are ignored. I have tried to limit to only one uncovered project and set the log level to All, the message is that the project is missing the PDB/MDB, however:
- I have checked the PDB file is presented in both tests and tested project
- Search directories are set correctly and point to the right directories
- PDB is set to full
- OpenCover report says that the project is missing PDB/MDB, that is why it is skipped

I am using the "path64" register switch.
I have started building one of the projects from scratch and the coverage was calculated correctly until I have added more dependencies to the solution projects, they also have their own dependencies and tests projects, etc. And the interesting thing here is that coverage is calculated for the related projects of the current, but not for the current project(missing PDB in logs, but it is in the bin folder). I will try to build a solution with a similar structure and put it in a public repository, this will consume some time.
I will invite you to my private repo for faster troubleshooting.

### Steps to reproduce the problem:

- [x] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf) again because we know you probably didn't do it the first time.
- [x] sample code or link to repository/gist is available (support is much, much faster this way)

**Comment by sawilde** [comment]
@slapikv 

I built the code and I could repeat your issue. At first glance the issue looks like a shadow copy issue.

I am not a user of cake however I edited the `tasks.tests.unit.cake` file that is downloaded as part of cake and added the following setting `ShadowCopy = false` to the `XUnit2Settings` and now all the assemblies record coverage. 

I don't know how you will fix this as the file is not under you source control - perhaps there are additional overrides you have access to - as I said I am not a cake user.

I hope this helps
